clients/src/main/java/org/apache/kafka/common/Node.java

```diff
@@ -102,12 +137,19 @@ public boolean equals(Object obj) {
             return false;
         if (port != other.port)
             return false;
+        if (rack == null) {
+            if (other.host != null)
```

Comment:
you mean other.rack?

Also, I'm wondering whether we want to consider a node to be a different node when the rack changed. I guess it depends on how it is used - except it isn't used...